bitnami/mongodb/templates/_helpers.tpl

```diff
@@ -346,9 +346,13 @@ Validate values of MongoDB&reg; exporter URI string - auth.enabled and/or tls.en
 */}}
 {{- define "mongodb.mongodb_exporter.uri" -}}
     {{- $uriTlsArgs := ternary "tls=true&tlsCertificateKeyFile=/certs/mongodb.pem&tlsCAFile=/certs/mongodb-ca-cert" "" .Values.tls.enabled -}}
-    {{- $uriAuth := ternary "root:$(echo $MONGODB_ROOT_PASSWORD | sed -r \"s/@/%40/g;s/:/%3A/g\")@" "" .Values.auth.enabled -}}
-
-    {{- printf "mongodb://%slocalhost:27017/admin?%s" $uriAuth $uriTlsArgs -}}
+    {{- if and .Values.metrics.username .Values.metrics.password }}
```

Comment:
What if metrics.password is empty? We may just enter the root user branch.

Reply:
I don't see how it would work with an empty password? The whole logic would break, since the creation of the secret is just executed when the password is defined and then it would try mount a non-existing variable. 
So maybe the best solution would be to trigger a warning that both have to be defined?

Reply:
I imagine that mounting the secret should be done only if the password is defined. I think it is potentially possible to access without password, if I'm not mistaken.

Reply:
Hi @javsalgar 
I just gave it a try. When I try to create a MongoDB user without a password I get the following error:
```
> db.getSiblingDB("admin").createUser({
...     user: "testmetrics",
...     pwd: "",
...     roles: [
...         { role: "clusterMonitor", db: "admin" },
...         { role: "read", db: "local" }
...     ]
... })
uncaught exception: Error: couldn't add user: User passwords must not be empty :
_getErrorWithCode@src/mongo/shell/utils.js:25:13
DB.prototype.createUser@src/mongo/shell/db.js:1386:11
@(shell):1:1
```
I think passwordless admin access is possible when you turn off authentication but I don't see a way how it can be possible for other users.

Reply:
Then it's fine by me to force the password for the metrics, which should be auto-generated just like the rest of passwords if not set.

Reply:
I added the automatic password generation if `.Values.metrics.username` is set but `.Values.metrics.password` is not. Together with the changes in the image this user should get automatically created now if defined. https://github.com/bitnami/bitnami-docker-mongodb/pull/285